forum/users/tests/test_token_serializer.py

```diff
@@ -0,0 +1,123 @@
+from unittest.mock import Mock
+from django.contrib.auth import get_user_model
+from rest_framework.test import APITestCase
+from rest_framework.exceptions import ValidationError
+from rest_framework_simplejwt.tokens import RefreshToken
+from users.models import Role
+from users.serializers import CustomTokenObtainPairSerializer
+
+User = get_user_model()
+
+
+class TestCustomTokenObtainPairSerializer(APITestCase):
+    def setUp(self):
+        # Create a test user
+        self.user = User.objects.create_user(
+            email="testuser@example.com",
+            password="securepassword",
+            role=Role.INVESTOR.value  # Assign a default role
+        )
+
+    def test_valid_role(self):
+        """Test that a valid role is added to the token payload."""
+        request_data = {"email": self.user.email, "password": "securepassword", "role": Role.STARTUP.value}
+        context = {"request": self._get_mock_request(request_data)}
+
+        serializer = CustomTokenObtainPairSerializer(context=context)
+        validated_data = serializer.validate(request_data)
+
+        # Extract refresh token
+        refresh_token = RefreshToken(validated_data["refresh"])
+
+        self.assertEqual(refresh_token["role"], Role.STARTUP.value)
+
+    def test_invalid_role(self):
+        """Test that an invalid role raises a ValidationError."""
+        request_data = {"email": self.user.email, "password": "securepassword", "role": 9999}  # Invalid role
+        context = {"request": self._get_mock_request(request_data)}
+
+        serializer = CustomTokenObtainPairSerializer(context=context)
+        with self.assertRaises(ValidationError) as exc:
+            serializer.validate(request_data)
```

Comment:
When checking for exceptions, you can use `assertRaisesMessage` to ensure the exception message is as expected.
```
with self.assertRaisesMessage(ValidationError, "Invalid role specified."):
    serializer.validate(request_data)
```